source/extensions/transport_sockets/tls/context_manager_impl.cc

```diff
@@ -93,14 +72,22 @@ absl::optional<uint64_t> ContextManagerImpl::secondsUntilFirstOcspResponseExpire
 }
 
 void ContextManagerImpl::iterateContexts(std::function<void(const Envoy::Ssl::Context&)> callback) {
-  for (const auto& ctx_weak_ptr : contexts_) {
-    Envoy::Ssl::ContextSharedPtr context = ctx_weak_ptr.lock();
+  for (const auto& context : contexts_) {
     if (context) {
       callback(*context);
     }
   }
 }
 
+void ContextManagerImpl::removeContext(const std::shared_ptr<Envoy::Ssl::Context>& old_context) {
```

Comment:
Use `ContextSharedPtr` here, and throughout

Reply:
fixed in https://github.com/envoyproxy/envoy/pull/19971/commits/77243d91d46f73e175ca10f81d48f6a7a53644fb